### PR TITLE
Controller Module

### PR DESCRIPTION
**Pull request by mithileshz**

### CODE REVIEW

File: MK94.CodeGenerator.Test/Attributes.cs [hunk 1]
```diff
@@ -7,13 +7,13 @@
 
 namespace MK94.CodeGenerator.Test
 {
-    public class ControllerFeature : ProjectAttribute
+    public class ControllersProject : ProjectAttribute
```
Comment (mukut1994) [inline MK94.CodeGenerator.Test/Attributes.cs:10]: Recommended naming is "_Feature"
Projects just implement or use features
please undo rename

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                if (!typeDef.Methods.Any())
+                    continue;
+
+                var file = codeGenerator.File($"{fileDef.Name}.g.cs");
+
+                file.WithUsing("Microsoft.AspNetCore.Mvc");
+
+                var ns = file.Namespace(project.NamespaceResolver(typeDef));
+
+                var typeName = typeDef.Type.Name.StartsWith('I') ? typeDef.Type.Name[1..] : typeDef.Type.Name;
```
Comment (mukut1994) [inline MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs:35]: use `string AsClassName(this TypeDefinition typeDef)` extension method instead

`var typeName = typeDef.AsClassName()`

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    public void AddTo(CSharpCodeGenerator codeGenerator)
+    {
+        foreach (var fileDef in project.Files)
+        {
+            foreach (var typeDef in fileDef.Types)
+            {
+                if (typeDef.Type.GetCustomAttribute<ControllerFeatureAttribute>() is null)
+                    continue;
+
+                if (!typeDef.Methods.Any())
```
Comment (mukut1994) [inline MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs:26]: swap order of ifs; better perf.

```
if (methods none) continue;

if (type not a controller feature) continue; // expensive call
```

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    if (getAttribute is not null)
+                        generatedMethod.Attribute(CsharpTypeReference.ToRaw("HttpGet"));
+
+                    var postAttribute = method.MethodInfo.GetCustomAttribute<PostAttribute>();
+
+                    if (postAttribute is not null)
+                        generatedMethod.Attribute(CsharpTypeReference.ToRaw("HttpPost"));
+                }
+
+                //var method = type.Method(MemberFlags.Public | MemberFlags.Override, CsharpTypeReference.ToType<string>(), "ToString");
```
Comment (mukut1994) [inline MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs:58]: delete

**Review by mukut1994 — CHANGES_REQUESTED** [review]

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs (hunk 3 above)
Comment (mithileshz) [reply MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs:26]: Done

File: MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs (hunk 2 above)
Comment (mithileshz) [reply MK94.CodeGenerator/Intermediate/CSharp/Modules/ControllerModule.cs:35]: done

**Review by mukut1994 — APPROVED** [review]

**Pull request merged**